src/service/resources-impl.js

```diff
@@ -1685,7 +1685,17 @@ export class Resource {
    * @return {boolean}
    */
   renderOutsideViewport() {
-    return this.element.renderOutsideViewport();
+    const renders = this.element.renderOutsideViewport();
+    // Boolean interface, element is either always allowed or never allowed to
+    // render outside viewport.
+    if (renders === true || renders === false) {
+      return renders;
+    }
+    // Numeric interface, element is allowed to render outside viewport when it
+    // is within X times the viewport height of the current viewport.
+    const viewportBox = this.viewport_.getRect();
+    const distanceFromViewport = this.layoutBox_.top - viewportBox.bottom;
+    return distanceFromViewport <= renders * viewportBox.height;
```

Comment:
This definitely looks pretty good. One request - could you please take a look in the code where we evaluate the direction of scrolling? We need to see if it'd be useful here as well. At the very least they shouldn't conflict.